### PR TITLE
Feat: Call POST /tokens with the element data

### PR DESCRIPTION
**Pull request by amkera**
## Description

-This PR adds a `createToken` function that calls a `POST` method to the `/tokens` endpoint with the request body of a token.

Test coverage is in this [PR](https://github.com/Basis-Theory/basistheory-ios-example/pull/7) inside the basistheory-ios-example repo.

Note: New [PR](https://github.com/Basis-Theory/basistheory-ios-example/pull/8) for additional changes to the tests after the one above was merged.

[Linear Ticket](https://linear.app/basis-theory/issue/ENG-3858/call-post-tokens-with-the-element-data)

**Note**: I capitalized the branch name by mistake, and lowercasing the branch name is the convention. I tried to change the name of the branch but couldn't. Please note the branch name is capitalized in case you try to check this branch out!

## Testing required outside of automated testing?

- [x] Not Applicable

### Screenshots (if appropriate):

- [ ] Not Applicable

## Rollback / Rollforward Procedure

- [x] Roll Forward
- [ ] Roll Back

## Reviewer Checklist

- [ ] Description of Change
- [ ] Description of outside testing if applicable.
- [ ] Description of Roll Forward / Backward Procedure
- [ ] Documentation updated for Change

### CODE REVIEW

File: BasisTheoryElements/Sources/BasisTheoryElements/BasisTheoryElements.swift [hunk 1]
```diff
@@ -55,6 +55,29 @@ final public class BasisTheoryElements {
...
     }
+    
+    public static func createToken(body: CreateTokenRequest, apiKey: String, completion: @escaping ((_ data: CreateTokenResponse?, _ error: Error?) -> Void)) -> Void {
+        var mutableBody = body
+        var mutableData = body.data?.value as! [String: Any]
+        replaceElementRefs(body: &mutableData)
+    
+        mutableBody.data = AnyCodable(mutableData)
+        
+        let apiKeyForTokenize = !BasisTheoryElements.apiKey.isEmpty ? BasisTheoryElements.apiKey : apiKey
```
Comment (brigonzalez) [inline BasisTheoryElements/Sources/BasisTheoryElements/BasisTheoryElements.swift:66]: we can make use of the `getApiKey` up above here!

**Comment by amkera** [comment]
Is one of the tests not passing since this is running against the `master` branch of the ios-example repo, which doesn't include my changes in one the tests?

**Comment by brigonzalez** [comment]
@amkera yup that seems to be it. let's get a PR out for those changes

**Comment by amkera** [comment]
> @amkera yup that seems to be it. let's get a PR out for those changes

Forgot to tag you, here it is: https://github.com/Basis-Theory/basistheory-ios-example/pull/8/files

**Review by brigonzalez — APPROVED** [review]

**Pull request merged**